g3/generate/pr/messages/creator.py

```diff
@@ -1,26 +1,50 @@
+from typing import Optional
+
 from g3.domain.message_tone import MessageTone
 from g3.generate.client import OpenAIChat
 from g3.generate.pr.prompts.creator import Creator as PromptCreator
 from g3.generate.preview.cli import Presenter
+from g3.git.client import get_commit_messages
 from g3.github.client import Client as GHClient
 
 
 class Creator:
-    def __init__(self):
+    def __init__(self, pr_id: Optional[int] = None):
```

Comment:
I would pass the `pr_id` as an `edit: int` parameter in the create method instead of the creator's initializer so that the initialization could be done eagerly once too.

Reply:
Maybe the commit edit needs the same as well.